src/Disks/StoragePolicy.cpp

```diff
@@ -56,13 +57,39 @@ StoragePolicy::StoragePolicy(
         config.keys(volumes_prefix, keys);
     }
 
+    std::unordered_set<UInt64> volume_priorities;
```

Comment:
std::set may be more natural, it allows to avoid min_element and max_element

Reply:
well, but it will be slower in this case -- two linear finds vs n logarithmic inserts

Reply:
Algorithmically speaking you are right.
In reality constructing of unordered_set is slower, insert is not free and traversing is slower as well (although it is O(n)).
If we are really curious we can do a perf test ;)